src/Services/GameDataService.cs

```diff
@@ -33,39 +36,61 @@ public class GameDataService {
         ctx.SaveChanges();
         return true;
     }
-
-    public GameDataSummary GetGameData(Viking viking, int gameId, bool isMultiplayer, int difficulty, int gameLevel, string key, int count, bool AscendingOrder, bool buddyFilter, string apiKey, DateTime? startDate = null, DateTime? endDate = null) {
-        // TODO: Buddy filter
-        List<GameDataResponse> selectedData;
-        IQueryable<Model.GameData> query = ctx.GameData.Where(x => x.GameId == gameId && x.IsMultiplayer == false && x.Difficulty == difficulty && x.GameLevel == gameLevel);
-
-        if (startDate != null && endDate != null)
-            query = query.Where(x => x.DatePlayed >= startDate.Value.ToUniversalTime() && x.DatePlayed <= endDate.Value.AddMinutes(2).ToUniversalTime());
-
-        var query2 = query.SelectMany(e => e.GameDataPairs)
+	
+	List<GameDataResponse> GameDataResponseToList(IQueryable<Model.GameData> query, string key, int count, bool AscendingOrder, string apiKey) {
```

Comment:
please do not mix space and tabs in indentation (and keep "standard" used in a specific file)